lib/src/spotify_base.dart

```diff
@@ -51,27 +50,26 @@ abstract class SpotifyApiBase {
     _shows = Shows(this);
   }
 
-  SpotifyApiBase(SpotifyApiCredentials credentials,
-      [http.BaseClient httpClient])
-      : this.fromClient(_getOauth2Client(credentials, httpClient));
-
-  SpotifyApiBase.fromAuthCodeGrant(
-      oauth2.AuthorizationCodeGrant grant, String responseUri)
-      : this.fromClient(grant.handleAuthorizationResponse(
-            Uri.parse(responseUri).queryParameters));
-
-  static oauth2.AuthorizationCodeGrant authorizationCodeGrant(
-      SpotifyApiCredentials credentials, http.BaseClient httpClient) {
-    return oauth2.AuthorizationCodeGrant(
-        credentials.clientId,
-        Uri.parse(SpotifyApiBase._authorizationUrl),
-        Uri.parse(SpotifyApiBase._tokenUrl),
-        secret: credentials.clientSecret,
-        httpClient: httpClient);
+  SpotifyApiBase(SpotifyApiCredentials credentials, [http.BaseClient httpClient, Function(SpotifyApiCredentials) callBack]) : this.fromClient(_getOauth2Client(credentials, httpClient, callBack));
```

Comment:
same here with indicative parameter name and those lines are too long, reformat them